In [44]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier
from sklearn.datasets import load_iris

In [2]:
iris = load_iris()

In [3]:
dir(iris)

['DESCR',
 'data',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

In [5]:
iris.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

In [6]:
x_train, x_test, y_train, y_test = train_test_split(iris.data, iris.target, test_size = 0.2)

In [8]:
len(x_train)

120

In [10]:
l = LogisticRegression()
l.fit(x_train, y_train)
l.score(x_test, y_test)

0.9666666666666667

In [11]:
from sklearn.ensemble import RandomForestClassifier

In [19]:
rf = RandomForestClassifier()
rf.fit(x_train, y_train)
rf.score(x_test, y_test)

0.9666666666666667

In [20]:
svm = SVC()
svm.fit(x_train, y_train)
svm.score(x_test, y_test)

0.9333333333333333

In [33]:
from sklearn.model_selection import cross_val_score
import numpy as np

In [73]:
s1 = cross_val_score(SVC(kernel= 'linear'), iris.data, iris.target, cv = 3)
np.average(s1)

0.9933333333333333

In [67]:
s2 = cross_val_score(RandomForestClassifier(n_estimators=40),iris.data, iris.target, cv = 3)
np.average(s2)

0.9666666666666667

In [65]:
s3 = cross_val_score(LogisticRegression(), iris.data, iris.target, cv = 3)
np.average(s3)

G:\ana\lib\site-packages\sklearn\linear_model\_logistic.py:763: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


0.9733333333333333

In [66]:
s4 = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target, cv = 3)
np.average(s4)

0.9666666666666667

   ### Best score is :     0.9733333333333333 of logistic regression

### If we use Linearn SVM best score is : 0.9933333333333333  SVM